lib/fluent/plugin/parser_csv.rb

```diff
@@ -27,13 +27,74 @@ class CSVParser < Parser
       config_param :keys, :array, value_type: :string
       desc 'The delimiter character (or string) of CSV values'
       config_param :delimiter, :string, default: ','
+      desc 'The parser type used to parse CSV line'
+      config_param :parser_type, :enum, list: [:normal, :fast], default: :normal
+
+      def configure(conf)
+        super
+
+        @quote_char = '"'
+        if @parser_type == :fast
+          m = method(:parse_fast)
+          self.singleton_class.module_eval do
+            define_method(:parse, m)
+          end
+        end
+      end
 
       def parse(text, &block)
         values = CSV.parse_line(text, col_sep: @delimiter)
         r = Hash[@keys.zip(values)]
         time, record = convert_values(parse_time(r), r)
         yield time, record
       end
+
+      def parse_fast(text, &block)
+        r = parse_fast_internal(text)
+        time, record = convert_values(parse_time(r), r)
+        yield time, record
+      end
+
+      # CSV.parse_line is too slow due to initialize lots of object and
+      # CSV module doesn't provide the efficient method for parsing single line.
+      # This method avoids the overhead of CSV.parse_line for typical patterns
+      def parse_fast_internal(text)
+        record = {}
+        text.chomp!
```

Comment:
Is executing `chomp!` okay here? 
If row value is `a,b, , `(User expects that the last column is space), should we remain the value as it is?

```
p CSV.parse_line('a,b, ,c, ')  #=> ["a", "b", " ", "c", " "]
```

Reply:
`chomp!` don't remove space character.